### PR TITLE
Update secret procedure for Restore

### PR DESCRIPTION
**Pull request by the-redback**
This PR fixes k8sdb/operator#123.
Secret is not needed to pre-defined while restoring snapshot.

### CODE REVIEW

File: pkg/controller/create.go [hunk 1]
```diff
@@ -233,6 +246,27 @@ func (c *Controller) findSecret(secretName, namespace string) (bool, error) {
...
 }
 
+func (c *Controller) findRestoreSecret(postgres *tapi.Postgres) (*apiv1.SecretVolumeSource, error) {
+	snapshot, err := c.ExtClient.Snapshots(postgres.Namespace).Get(postgres.Spec.Init.SnapshotSource.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	srcPostgres, err := c.ExtClient.Postgreses(postgres.Namespace).Get(snapshot.Spec.DatabaseName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
```
Comment (aerokite) [inline pkg/controller/create.go]: Do not return error. May be Postgres is deleted. But secret is available. 
If Postgres is missing, Or Secret is missing, Create new secret from Snapshot object

Comment (aerokite) [reply pkg/controller/create.go]: Like this: https://github.com/k8sdb/postgres/blob/master/pkg/controller/postgres.go#L342

File: pkg/controller/postgres.go [hunk 2]
```diff
@@ -245,6 +245,12 @@ func (c *Controller) ensureStatefulSet(postgres *tapi.Postgres) error {
 		return err
 	}
 
+	_postgres, err := c.ExtClient.Postgreses(postgres.Namespace).Get(postgres.Name, metav1.GetOptions{})
```
Comment (aerokite) [inline pkg/controller/postgres.go]: postgres, err := c.ExtClient.Postgreses(postgres.Namespace).Get(postgres.Name, metav1.GetOptions{})

Use this

File: hack/docker/postgres/9.5/util/Dockerfile [hunk 3]
```diff
@@ -8,7 +8,7 @@ FROM appscode/python:2.7
 ## postgres client major version
 ENV PG_MAJOR 9.5
 ## osm version
-ENV OSM_VERSION 0.5.0
+ENV OSM_VERSION 0.6.0
```
Comment (tamalsaha) [inline hack/docker/postgres/9.5/util/Dockerfile]: Revert this. We should not change existing image.

File: test/e2e/postgres_test.go [hunk 4]
```diff
@@ -215,7 +215,7 @@ var _ = Describe("Postgres", func() {
 					}
 				})
 
-				It("should take Snapshot successfully", shouldTakeSnapshot)
+				FIt("should take Snapshot successfully", shouldTakeSnapshot)
```
Comment (tamalsaha) [inline test/e2e/postgres_test.go]: `FIt` -> `It`

**Pull request merged**